serverless/openvino/omz/intel/person-reidentification-retail-300/nuclio/reid.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (C) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import math
+import numpy
+
+from scipy.optimize import linear_sum_assignment
+from scipy.spatial.distance import euclidean, cosine
+
+from model_loader import ModelLoader
+
+class ReID:
+    def __init__(self, model_xml, model_bin):
+        self.model = ModelLoader(model_xml, model_bin)
+
+    def infer(self, image0, boxes0, image1, boxes1, threshold, distance):
+        similarity_matrix = self._compute_similarity_matrix(image0,
+            boxes0, image1, boxes1, distance)
+        row_idx, col_idx = linear_sum_assignment(similarity_matrix)
+        results = [-1] * len(boxes0)
+        for idx0, idx1 in zip(row_idx, col_idx):
+            if similarity_matrix[idx0, idx1] <= threshold:
+                results[idx0] = idx1
+
+        return results
+
+    def _match_boxes(self, box0, box1, distance):
+        cx0 = (box0["points"][0] + box0["points"][2]) / 2
+        cy0 = (box0["points"][1] + box0["points"][3]) / 2
+        cx1 = (box1["points"][0] + box1["points"][2]) / 2
+        cy1 = (box1["points"][1] + box1["points"][3]) / 2
+        is_good_distance = euclidean([cx0, cy0], [cx1, cy1]) <= distance
+        is_same_label = box0["label_id"] == box1["label_id"]
+
+        return is_good_distance and is_same_label
+
+    def _match_crops(self, crop0, crop1):
+        embedding0 = self.model.infer(crop0)
```

Comment:
Codacy found an issue: [No value for argument 'threshold' in method call](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5729179)